src/rsa/blinding.rs

```diff
@@ -0,0 +1,585 @@
+            ($i:expr, $l:expr) => {
+                {
+                    let x = [$l];
+                    let p = select_sampling_params(&x[..]);
+                    let mask = Limb::max_value() >> (LIMB_BITS - 1 - $i);
+                    assert!(!p.extend_limbs_by_one);
+                    assert!(!p.reduce_when_over_bound);
+                    assert_eq!(mask, p.most_sig_limb_mask);
+                }
+            }
+        }
+
+        for i in 0..(LIMB_BITS - 2) {
+            let l = 1 << i;
+
+            assert_normal!(i, l);
+            assert_normal!(i, l | 1);
+            assert_normal!(i, l | l >> 3);
+        }
+
+        for i in 0..LIMB_BITS {
+            let l = 1 << i;
+
+            assert_normal!(i, l | l >> 1);
+            assert_normal!(i, l | l >> 2);
+            assert_normal!(i, Limb::max_value() >> (LIMB_BITS - 1 - i));
+        }
+    }
+
+    #[test]
+    fn test_limbs_in_range() {
+        use super::{SamplingParams,Range};
+
+        let params = SamplingParams {
+            most_sig_limb_mask: Limb::max_value(),
+            reduce_when_over_bound: false,
+            extend_limbs_by_one: false,
+        };
+
+        let limbs = &[Limb::max_value(), Limb::max_value()];
+        let range = Range { max_exclusive: limbs, sampling_params: &params };
+        assert!(!range.are_limbs_within(&[Limb::max_value(),
+                                          Limb::max_value()]));
+        assert!(range.are_limbs_within(&[Limb::max_value(),
+                                         Limb::max_value() - 1]));
+        assert!(range.are_limbs_within(&[Limb::max_value() - 1,
+                                         Limb::max_value()]));
+        assert!(!range.are_limbs_within(&[0, 0]));
+        assert!(range.are_limbs_within(&[1, 0]));
+        assert!(range.are_limbs_within(&[0, 1]));
+
+        let limbs = &[0x12345678, 0xdeadbeef];
+        let range = Range { max_exclusive: limbs, sampling_params: &params };
+        assert!(!range.are_limbs_within(&[0x12345678, 0xdeadbeef]));
+        assert!(range.are_limbs_within(&[0x12345678 - 1, 0xdeadbeef]));
+        assert!(range.are_limbs_within(&[0x12345678, 0xdeadbeef - 1]));
+        assert!(!range.are_limbs_within(&[0x12345678 + 0x10, 0xdeadbeef]));
+        assert!(!range.are_limbs_within(&[0x12345678, 0xdeadbeef + 0x10]));
+
+        let limbs = &[0, 1];
+        let range = Range { max_exclusive: limbs, sampling_params: &params };
+        assert!(!range.are_limbs_within(&[0, 0]));
+        assert!(range.are_limbs_within(&[1, 0]));
+        assert!(!range.are_limbs_within(&[0, 1]));
+        assert!(range.are_limbs_within(&[Limb::max_value(), 0]));
+
+        let limbs = &[2];
+        let range = Range { max_exclusive: limbs, sampling_params: &params };
+        assert!(!range.are_limbs_within(&[0]));
+        assert!(range.are_limbs_within(&[1]));
+        assert!(!range.are_limbs_within(&[2]));
+    }
+
+    #[test]
+    #[allow(unused_results)]
```

Comment:
Let's not use `allow(unused_results)`, even in tests. Instead we can use the `let _ = ` pattern.